File: python/grid.py
import numpy as np

class NozzleGeom:
    """
    1D nozzle geometry and grid
    """

    def __init__(self, params, domain ):
        x = np.linspace(params.x0, params.x1, params.Imax + 1)

        self.x = x[domain.slice]

        # piecewise linear area gradient
        self.dAdx = np.where(self.x <= 0.0, -0.0318, 0.0510)

        # area distribution
        self.A = params.A_throat + self.dAdx * self.x

File: cpp/grid.cpp
#include "grid.hpp"

// =============================================
// Constructor
// =============================================

NozzleGrid::NozzleGrid()
{
    const int N = Parameters::Imax + 1;

    x.resize(N);
    A.resize(N);
    dAdx.resize(N);

    // -----------------------------------------
    // Grid generation
    // -----------------------------------------

    for(int i = 0; i < N; ++i)
    {
        x[i] = Parameters::x0 + i * Parameters::dx;
    }

    // -----------------------------------------
    // Geometry
    // -----------------------------------------

    for(int i = 0; i < N; ++i)
    {
        if(x[i] <= 0.0)
        {
            dAdx[i] = -0.0318;
        }
        else
        {
            dAdx[i] = 0.0510;
        }

        A[i] = Parameters::A_throat + dAdx[i] * x[i];
    }
}